src/components/Dashboard/Category-Card/AdminModal.js

```diff
@@ -0,0 +1,193 @@
+import React, { useState } from "react";
+import { Modal, Button, Form } from "react-bootstrap";
+
+const AdminModal = (props) => {
+  const [airmanId, setAirmanId] = useState("");
+  const [flightId, setFlightId] = useState("");
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [pt, setPT] = useState("Current");
+  const [evaluation, setEval] = useState("Current");
+  const [adls, setADLS] = useState("Current");
+  const [medical, setMedical] = useState("Current");
+
+  const createUrl = () => {
+    const key = props.optionname.toLowerCase();
+    console.log(key);
+    if (props.optionname === "Add") {
+      return `https://us-central1-cc-dashboard-afb15.cloudfunctions.net/${key}Airman/${flightId}/`;
+    } else {
+      return `https://us-central1-cc-dashboard-afb15.cloudfunctions.net/${key}Airman/${flightId}/${airmanId}`;
+    }
+  };
+
+  const getMethod = () => {
+    switch (props.optionname) {
+      case "Add":    return "POST";
+      case "Remove": return "DELETE";
+      case "Update": return "PUT";
+
+      default: return "bad-option";
+    }
+  };
+
+  const getBody = () => {
+    return props.optionname !== "Remove"
+      ? {
+          method: getMethod(),
+          mode: "cors",
+          accept: "*/*",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            firstName: firstName,
+            lastName: lastName,
+            adls: adls,
+            pt: pt,
+            medical: medical,
+            evaluation: evaluation,
+          }),
+        }
+      : {
+          method: getMethod(),
+          mode: "cors",
+          accept: "*/*",
+        };
+  };
+
+  return (
```

Comment:
This returns a LOT of code.  We could almost certainly break this into separate components that could be imported here.  That might allow us to simplify the return in this file (and the other subcomponents would have comparatively simple returns), which could save a lot of time during any refactoring or troubleshooting of the app in the future.